src/sentry/static/sentry/app/views/eventsV2/utils.tsx

```diff
@@ -423,10 +436,83 @@ export function getExpandedResults(
     }
   });
 
-  // Remove fields from view largest index to smallest to not
-  // disturbe lower indexes until the end.
-  fieldsToRemove.reverse().forEach((i: number) => {
-    nextView = nextView.withDeletedColumn(i, undefined);
+  const transformedFields = new Set();
+  const fieldsToDelete: number[] = [];
+
+  // make a best effort to transform aggregated columns with its non-aggregated form
+  fieldsToUpdate.forEach((indexToUpdate: number) => {
+    const currentField: FieldType = nextView.fields[indexToUpdate];
+    const exploded = explodeField(currentField);
+
+    // check if we can use an aggregated transform function
+
+    for (const tuple of TRANSFORM_AGGREGATES) {
```

Comment:
If TRANSFORM_AGGREGATES was a lookup of fields => transforms you could skip a layer of loops. :golf: